aiidalab_widgets_base/databases.py

```diff
@@ -58,6 +59,14 @@ def __init__(self, **kwargs):
         ]
         super(CodQueryWidget, self).__init__(children=children, **kwargs)
 
+    @observe('manager')
+    def _change_manager(self, value):
+        """Set structure manager trait."""
+        manager = value['new']
+        if manager is None:
+            return
+        dlink((self, 'structure'), (manager, 'input_structure'))
```

Comment:
My motivation here is the following. I would like to give to the importer/editor developers a possibility to access any element of the manager if needed. For the `editors` the situation is quite clear to me: when editing a structure one wants to have the freedom to sync any useful trait of the viewer, be it camera position or currently selected atom. Also, this approach looks more future-proof to me. If a developer realizes that they need to link some additional trait they can just do it without a need to change the behaviour of the `StructureManagerWidget`.

With the importers, however, the situation appears less clear. It is true that currently, it is enough to only link the`input_structure` trait. Nevertheless, I don't know what might be needed in future, and, I think it would be nice to manage importers and editors in a similar way.

So I would really prefer to give to the importers/editors link to the StructureManager object and let them decide which traits they want to sync.

Reply:
This appears like premature generalization and a leak of responsibilities to me. Specifically, having this bidirectional dependency is going to make it extremely difficult to have a clear separation of concerns between the manager and the various importers/editors in the future. In conclusion, I am strongly opposed to this.

Reply:
I replaced the `manager` trait with `connect_to_manager()` method as we discussed offline.